### DISCUSSION TITLE
Subcommand Handling

### DISCUSSION THREAD

**Discussion by eklatzer** [postmodern/command_mapper.rb#13] [Q&A] [answered]
Hey, I am using `command_mapper 0.3.1` using subcommands. I did not find any documentation about the usage of subcommands, except the definition of a command with subcommands here:
https://github.com/postmodern/command_mapper.rb/blob/main/lib/command_mapper/command.rb#L502

This is the implementation I currently have:
```ruby
require 'command_mapper/command'

class Git < CommandMapper::Command
  command "git" do
    subcommand "log" do
      option "--max-count", value: {type: Num.new()}
    end
  end
end

Git.run({ :log => { :max_count => 5 }})
```

Is this how it is intended to be done? Are there any examples showcasing the usage of subcommands?

Thanks in advance

**Comment by postmodern** [comment, marked as answer]
Yes that is how you invoke a subcommand, by passing a nested Hash of options into the outer-command. All of the same execution methods can be used (ex: `.run`, `.spawn`, `.capture`, `.popen`, `.sudo`) to execute subcommands. Off the top of my head, there's [ruby-amass](https://github.com/postmodern/ruby-amass#examples) which defines subcommands for the [amass](https://github.com/OWASP/Amass) CLI.

**Reply by eklatzer** [reply]
Thanks, would be nice to have an example in the docs.

**Reply by postmodern** [reply]
How does this look? https://github.com/postmodern/command_mapper.rb#defining-sub-commands
